apps/condo/domains/ticket/access/TicketCommentFile.js

```diff
@@ -51,7 +66,38 @@ async function canReadTicketCommentFiles ({ authentication: { item: user }, cont
     }
 }
 
-const checkManageCommentFileAccess = async ({ user, operation, originalInput, itemId, context }) => {
+const checkManageCommentFileAccess = async (args) => {
+    const { authentication: { item: user }, operation, originalInput, itemId, context } = args
+
+    if (user.type === SERVICE) {
+        const hasAccess = await canManageObjectsAsB2BAppServiceUser(args)
+        if (!hasAccess) return false
+
+        if (operation === 'create') {
+            const ticketCommentFromOriginalInput = get(originalInput, ['ticketComment', 'connect', 'id'])
+            if (!ticketCommentFromOriginalInput) return false
+
+            const ticketComment = await getByCondition('TicketComment', {
+                id: ticketCommentFromOriginalInput,
+                deletedAt: null,
+            })
+            if (!ticketComment) return false
+
+            return ticketComment.createdBy === user.id
+        } else if (operation === 'update') {
+            const ticketCommentFile = await getByCondition('TicketCommentFile', {
+                id: itemId,
```

Comment:
What do you expect here? If `itemId` is `undefined` (in case of bulk requests), filter should probably be dropped?

Reply:
How about implementing something similar I've shared above?